crypto/totp.go

```diff
@@ -98,24 +98,32 @@ func TOTPCompare(key string, period time.Duration, adjust int, length int, chall
 
 // TOTPCompareWithVariance the expected TOTP calculation with the challenge in constant time. If variance is > 0
 // constant time execution is not guaranteed, allows for totp to fall with the variance range of steps + or -
-func TOTPCompareWithVariance(key string, period time.Duration, length int, variance uint, challenge string) (bool, error) {
+func TOTPCompareWithVariance(key string, period time.Duration, length int, variance uint, challenge string) (ok bool, err error) {
+	ok, _, err = TOTPCompareAndGetDrift(key, period, length, variance, challenge)
+	return
+}
+
+// TOTPCompareAndGetDrift the expected TOTP calculation with the challenge in constant time. If variance is > 0
+// constant time execution is not guaranteed, allows for totp to fall with the variance range of steps + or -, variance is returned
+func TOTPCompareAndGetDrift(key string, period time.Duration, length int, variance uint, challenge string) (bool, int, error) {
 	var eq bool
 	var err error
 	// do 0, +1 && -1, +2 && -2, etc.
 	for vary := 0; vary <= int(variance); vary++ {
```

Comment:
we want to skew this using the drift passed in, so if we have the values we would want to check the frames according to this:
```
drift   variance  frames
0       0              0
0       1              -1,0,1
0       2              -2,-1,0,1,2
1       1              0,1,2
1       2              -1,0,1,2,3
-1      1              -2,-1,0
-1      2              -3,-2,-1,0,1
```